homeassistant/components/remootio/__init__.py

```diff
@@ -0,0 +1,105 @@
+"""The Remootio integration."""
+from __future__ import annotations
+
+import logging
+
+from aioremootio import ConnectionOptions, RemootioClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ENTITY_ID, ATTR_NAME, CONF_HOST, Platform
+from homeassistant.core import HomeAssistant, callback
+
+from .const import (
+    ATTR_SERIAL_NUMBER,
+    ATTR_TYPE,
+    CONF_API_AUTH_KEY,
+    CONF_API_SECRET_KEY,
+    CONF_SERIAL_NUMBER,
+    DOMAIN,
+    EVENT_HANDLER_CALLBACK,
+    EVENT_TYPE,
+    REMOOTIO_CLIENT,
+)
+from .cover import RemootioCoverEvent
+from .utils import create_client
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.COVER]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Remootio from a config entry."""
+
+    _LOGGER.debug("Doing async_setup_entry. entry [%s]", entry.as_dict())
+
+    @callback
+    def handle_event(event: RemootioCoverEvent) -> None:
+        _LOGGER.debug(
+            "Firing event. EvenType [%s] RemootioCoverEntityId [%s] RemootioDeviceSerialNumber [%s]",
+            event.type,
+            event.entity_id,
+            event.device_serial_number,
+        )
+
+        hass.bus.async_fire(
+            EVENT_TYPE,
+            {
+                ATTR_ENTITY_ID: event.entity_id,
+                ATTR_SERIAL_NUMBER: event.device_serial_number,
+                ATTR_NAME: event.entity_name,
+                ATTR_TYPE: event.type,
+            },
+        )
```

Comment:
Please remove the custom event for now, this functionality should be modeled as an event entity instead, in a follow-up PR: https://developers.home-assistant.io/docs/core/entity/event

Reply:
How should I then fire events based on the events that the Remootio fires and on which the user may want to set up an automation, e.g. if the controlled garden gate or garage door remains open for a time that can be configured on the Remootio?

Reply:
Please look into event entities which I linked to: <https://developers.home-assistant.io/docs/core/entity/event/>, I think it will fit this use case very well.